doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -1,74 +1,176 @@
+the type they represent.
+
+.. coqtop:: all
+
+   intros.
+
+Some tactics, such as :tacn:`split`, create new goals, which may
+be referred to as :gdef:`subgoals <subgoal>` in descriptions for clarity.
+Goals are numbered from 1 to N at each step of the proof to permit applying a
+tactic to specific goals.
+
+.. coqtop:: all
+
+   split.
+
+"Variables" may refer specifically to facts for which the type of their type
+is `Set` or `Type`, and "hypotheses" to :term:`propositions <proposition>`,
+which are facts for which the type of their type is `Prop` or `SProp`,
+but these terms are also used interchangeably.
+
+.. coqtop:: out
+
+   let t_n := type of n in idtac "type of n :" t_n;
+   let tt_n := type of t_n in idtac "type of" t_n ":" tt_n.
+   let t_H := type of H in idtac "type of H :" t_H;
+   let tt_H := type of t_H in idtac "type of" t_H ":" tt_H.
+
+A proof script, consisting of the tactics that are applied to prove a
+theorem, is often informally referred to as a "proof".
+The real proof, whether complete or incomplete, is a term, the :gdef:`proof term`,
+which users may occasionally want to examine.  (This is based on the
+*Curry-Howard isomorphism* :cite:`How80,Bar81,Gir89,H89`, which is
+a correspondence between between proofs and terms and between
+propositions and types of λ-calculus.  The isomorphism is also
+sometimes called the "propositions-as-types correspondence")
+
+The :cmd:`Show Proof` command displays the incomplete proof term
+before you've completed the proof.  For example:
+
+.. coqtop:: all
+
+   Show Proof.
+
+The incomplete parts, the goals, are represented by
+:term:`existential variables <existential variable>`
+with names that begin with `?Goal`.  The :cmd:`Show Existentials` command
+shows each existential with the hypotheses and conclusion for the associated goal.
+
+.. coqtop:: all
+
+   Show Existentials.
+
+Coq's kernel verifies the correctness of proof terms when it exits
+proof mode by checking that the theorem statement is the same as the
+type of the proof term.
+
+After a proof is completed, :cmd:`Print` `<theorem_name>`
+shows the proof term and its type.  The type appears after
+the colon (`forall ...`), as for this theorem from Coq's standard library:
+
+.. coqtop:: all
+
+   Print proj1.
 
 .. _proof-editing-mode:
 
-Entering and leaving proof editing mode
----------------------------------------
+Entering and exiting proof mode
+-------------------------------
+
+Coq enters :term:`proof mode` when you begin a proof through
+commands such as :cmd:`Theorem` or :cmd:`Goal`.  Coq user interfaces
+usually have a way to indicate that you're in proof mode.
+
+Tactics are available only in proof mode (currently giving syntax
+error outside of proof mode).  Most commands can be used both in and out of
+proof mode, but some commands only work in proof mode.
+By convention, tactics begin with lowercase letters
+while commands begin with uppercase letters.
```

Comment:
Maybe add `:term:` for tactics and commands since they are defined elsewhere and maybe remove the last sentence since this information is already in the definition of these terms.